model/validation.go

```diff
@@ -204,16 +204,353 @@ func (t Tags) Validate() error {
+		retVal = multierror.Append(retVal, fmt.Errorf("throttle percent must be in range 0..100"))
+	}
+
+	if throttle.DownstreamLimitBps < 0 {
+		retVal = multierror.Append(retVal, fmt.Errorf("downstream_limit_bps invalid"))
+	}
+
+	if throttle.UpstreamLimitBps < 0 {
+		retVal = multierror.Append(retVal, fmt.Errorf("upstream_limit_bps invalid"))
+	}
+
+	if throttle.GetThrottleAfterSeconds() < 0 {
+		retVal = multierror.Append(retVal, fmt.Errorf("throttle_after_seconds invalid"))
+	}
+
+	if throttle.GetThrottleAfterBytes() < 0 {
+		retVal = multierror.Append(retVal, fmt.Errorf("throttle_after_bytes invalid"))
+	}
+
+	// TODO Check DoubleValue throttle.GetThrottleForSeconds()
+
+	return retVal
+}
+
+// ValidateLoadBalancing validates Load Balancing
+func ValidateLoadBalancing(lb *proxyconfig.LoadBalancing) error {
+	var retVal error
+
+	// Currently the policy is just a name, and we don't validate it
+
+	return retVal
+}
+
+// ValidateCircuitBreaker validates Circuit Breaker
+func ValidateCircuitBreaker(cb *proxyconfig.CircuitBreaker) error {
+	var retVal error
+
+	if simple := cb.GetSimpleCb(); simple != nil {
+		if simple.MaxConnections < 0 {
+			retVal = multierror.Append(retVal, fmt.Errorf("circuit_breaker max_connections must be in range [0..]"))
+		}
+		if simple.HttpMaxPendingRequests < 0 {
+			retVal = multierror.Append(retVal, fmt.Errorf("circuit_breaker max_pending_requests must be in range [0..]"))
+		}
+		if simple.HttpMaxRequests < 0 {
+			retVal = multierror.Append(retVal, fmt.Errorf("circuit_breaker max_requests must be in range [0..]"))
+		}
+		if simple.SleepWindowSeconds < 0 {
+			retVal = multierror.Append(retVal, fmt.Errorf("circuit_breaker sleep_window_seconds must be in range [0..]"))
+		}
+		if simple.HttpConsecutiveErrors < 0 {
+			retVal = multierror.Append(retVal, fmt.Errorf("circuit_breaker http_consecutive_errors must be in range [0..]"))
+		}
+		if simple.HttpDetectionIntervalSeconds < 0 {
+			retVal = multierror.Append(retVal,
+				fmt.Errorf("circuit_breaker http_detection_interval_seconds must be in range [0..]"))
+		}
+		if simple.HttpMaxRequestsPerConnection < 0 {
+			retVal = multierror.Append(retVal,
+				fmt.Errorf("circuit_breaker http_max_requests_per_connection must be in range [0..]"))
+		}
+		if simple.HttpMaxEjectionPercent < 0 || simple.HttpMaxEjectionPercent > 100 {
+			retVal = multierror.Append(retVal,
+				fmt.Errorf("circuit_breaker http_max_ejection_percent must be in range [0..100]"))
+		}
+	}
+
+	return retVal
+}
+
 // ValidateRouteRule checks routing rules
 func ValidateRouteRule(msg proto.Message) error {
+
 	value, ok := msg.(*proxyconfig.RouteRule)
 	if !ok {
-		return fmt.Errorf("Cannot cast to routing rule")
+		return fmt.Errorf("cannot cast to routing rule")
 	}
+	var retVal error
 	if value.Destination == "" {
-		return fmt.Errorf("RouteRule must have a destination service")
+		retVal = multierror.Append(retVal, fmt.Errorf("routeRule must have a destination service"))
```

Comment:
`routeRule` should not be lower cased. Also try avoid the name `RouteRule` since we may change that name at some point, and would be hard to hunt down all usages in strings.